src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -101,6 +101,34 @@ protected virtual SourceText CreateText(Stream stream, Workspace workspace)
             return textAndVersion;
         }
 
+        internal override TextAndVersion LoadTextAndVersionSynchronously(Workspace workspace, DocumentId documentId, CancellationToken cancellationToken)
+        {
+            DateTime prevLastWriteTime = FileUtilities.GetFileTimeStamp(_path);
+
+            TextAndVersion textAndVersion;
+
+            // Open file for reading with FileShare mode read/write/delete so that we do not lock this file.
+            using (var stream = FileUtilities.RethrowExceptionsAsIOException(() => new FileStream(_path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete, bufferSize: 4096, useAsync: false)))
+            {
+                var version = VersionStamp.Create(prevLastWriteTime);
+                var text = CreateText(stream, workspace);
```

Comment:
It's a shame to duplicate this whole method, though I don't see much alternative either.